qos-monitor/src/main/java/eu/arrowhead/core/qos/service/ping/monitor/impl/OrchestratedExternalPingMonitor.java

```diff
@@ -0,0 +1,270 @@
+	private UUID requestExternalMeasurement(final String address) {
+		logger.debug("requestExternalMeasurement started...");
+
+			try {
+				final IcmpPingRequestACK acknowledgedMeasurmentRequest = driver.requestExternalPingMonitorService(getPingMonitorProvidersServiceUri(), createIcmpPingRequest(address));
+				validateAcknowledgedMeasurmentRequest(acknowledgedMeasurmentRequest);
+
+				final UUID startedExternalMeasurementProcessId = acknowledgedMeasurmentRequest.getExternalMeasurementUuid();
+				logger.info("IcmpPingRequestACK received, with process id: " + startedExternalMeasurementProcessId);
+
+				return startedExternalMeasurementProcessId;
+
+			} catch (final Exception ex) {
+				logger.info(ex);
+
+				throw new ArrowheadException("External Ping Monitor is not available at: " + address );
+			}
+
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private void validateAcknowledgedMeasurmentRequest(final IcmpPingRequestACK acknowledgedMeasurmentRequest) {
+		logger.debug("validateAcknowledgedMeasurmentRequest started...");
+
+		try {
+			Assert.notNull(acknowledgedMeasurmentRequest, "IcmpPingRequestACK is null");
+			Assert.notNull(acknowledgedMeasurmentRequest.getAckOk(), "IcmpPingRequestACK.ackOk is null");
+			Assert.isTrue(acknowledgedMeasurmentRequest.getAckOk().equalsIgnoreCase("OK"), "IcmpPingRequestACK is null");
+		
+			
+		} catch (final Exception ex) {
+			logger.warn("External pingMonitorProvider replied invalid ack : " + ex);
+
+			throw new ArrowheadException("External pingMonitorProvider replied invalid ack", ex);
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private IcmpPingRequestDTO createIcmpPingRequest(final String address) {
+		logger.debug("createIcmpPingRequest started...");
+
+		final IcmpPingRequestDTO request = new IcmpPingRequestDTO();
+		request.setHost(address);
+		request.setPacketSize(pingMeasurementProperties.getPacketSize());
+		request.setTimeout(Long.valueOf(pingMeasurementProperties.getTimeout()));
+		request.setTimeToRepeat(pingMeasurementProperties.getTimeToRepeat());
+		request.setTtl(ICMP_TTL);
+
+		return request;
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private UriComponents getPingMonitorProvidersServiceUri() {
+		logger.debug("getPingMonitorProvidersServiceUri started...");
+
+		final SystemResponseDTO provider = cachedPingMonitorProvider.getProvider();
+		final ServiceSecurityType securityType = cachedPingMonitorProvider.getSecure();
+		final String path = cachedPingMonitorProvider.getServiceUri();
+
+		switch (securityType) {
+		case NOT_SECURE:
+			return createHttpUri(provider, path);
+		case CERTIFICATE:
+			return createHttpsUri(provider, path);
+		case TOKEN:
+			throw new InvalidParameterException("TOKEN security is not supported yet, for PingMonitor Providers. ");
```

Comment:
not needed, the default branch does the same

Reply:
fixed